src/main/java/bio/terra/service/filedata/google/firestore/FireStoreUtils.java

```diff
@@ -43,11 +45,14 @@ public class FireStoreUtils {
   // Firestore limits batches to 500
   public static final int MAX_FIRESTORE_BATCH_SIZE = 500;
 
-  private ConfigurationService configurationService;
+  private final ConfigurationService configurationService;
+  private final long futureTimoutSeconds;
 
   @Autowired
-  public FireStoreUtils(ConfigurationService configurationService) {
+  public FireStoreUtils(
+      ConfigurationService configurationService, ApplicationConfiguration configuration) {
```

Comment:
Not a blocker --

It would be nice to introduce a new `FirestoreConfiguration` record and refactor out the relevant configs from `ConfigurationService` and `ApplicationConfiguration` there.  The higher-level configs feel like catch-alls that are hard to navigate.  `ConfigurationService` I think exists to make testing overrides easier, but I suspect that's an old pattern and can be handled differently.

Reply:
Totally fair.  I do worry about the lift of having to also change the properties in helm but let me see if we override any of them. I can handle that in my next PR